src/Symfony/Component/Scheduler/Event/PreRunEvent.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Event;
+
+use Symfony\Component\Scheduler\Generator\MessageContext;
+use Symfony\Component\Scheduler\ScheduleProviderInterface;
+
+class PreRunEvent
+{
+    private bool $isCancel = false;
+
+    public function __construct(private readonly ScheduleProviderInterface $schedule, private readonly MessageContext $messageContext, private readonly object $message)
+    {
+    }
+
+    public function getMessageContext(): MessageContext
+    {
+        return $this->messageContext;
+    }
+
+    public function getSchedule(): ScheduleProviderInterface
+    {
+        return $this->schedule;
+    }
+
+    public function getMessage(): object
+    {
+        return $this->message;
+    }
+
+    public function shouldCancel(bool $shouldCancel = null): bool
```

Comment:
I'm not sure we have a lot of similar methods in the code base where one method is used for both the setter and the getter.

Reply:
yes, sorry. Everything is under hasBeenCancelled() now

Reply:
it's a duplicate of `shouldHandle()` of WorkerMessageReceivedEvent needed in the `PreRunEvent`